src/ExpressionEvaluator/Core/Source/ResultProvider/Expansion/MemberExpansion.cs

```diff
@@ -27,7 +27,16 @@ internal sealed class MemberExpansion : Expansion
             Predicate<MemberInfo> predicate,
             Formatter formatter)
         {
-            var runtimeType = value.Type.GetLmrType();
+            // For members of type DynamicProperty (part of Dynamic View expansion), we want
+            // to expand the underlying value (not the members of the DynamicProperty type).
+            var type = value.Type;
+            var isDynamicProperty = type.GetLmrType().IsDynamicProperty();
+            if (isDynamicProperty)
+            {
+                value = value.GetFieldValue("value", inspectionContext);
+            }
```

Comment:
Does this prevent debugging the actual 'DynamicProperty' implementation? Can the caller be explicit, say passing in an 'inDynamicView' bool?